suggested edits on README

jbondpdx opened the pull request:
Howdy!

As part of the work for the Puppet SDK, I'd like to submit some edits to the Puppet::Syntax README.  I've edited and reorganized the README with best docs practices in mind; of please let me know if I've got something wrong or you'd like me to make adjustments.

Pinging @DavidS for visibility (also, he helped review my changes before this PR).

jbondpdx commented on README.md:
```diff
@@ -2,136 +2,138 @@
+        PuppetSyntax.app_management = true
 
-Checks `.erb` files for syntax errors
+  The `app_management` setting is supported with Puppet 4.3 or greater and is off by default.
```
Add back in: With Puppet 5, `app_management` is always enabled.

DavidS commented on README.md:
```diff
@@ -2,136 +2,138 @@
 
 - Puppet >= 2.7 that provides the `validate` face.
-- Ruby >= 1.8 with `erb` from stdlib.
+- Ruby >= 1.8 with `erb` from `puppetlabs-stdlib`.
```
Errr. Yes. I missed that.

jbondpdx replied:
Oops, that makes sense, sorry!

jbondpdx commented:
Thank you for the review, @domcleal . I've updated my PR with those fixes.

I squashed the commits, as that's usually what I do at work. But truth be told, I haven't contributed to community projects before, so I don't really know whether squashing them is proper etiquette or not. :-)

domcleal commented on README.md:
```diff
@@ -2,136 +2,138 @@
+* To install with the Gemfile, add:
 
-Test all manifests and templates relative to your `Rakefile`:
+        $ gem 'puppet-syntax'
```
The `$ ` should be removed, it shouldn't be in a Gemfile and isn't a command.

domcleal commented on README.md:
```diff
@@ -2,136 +2,138 @@
+  And then execute:
 
-A non-zero exit code and error message will be returned for any failures:
+        $ bundle
```
`bundle install`

domcleal commented:
Thanks for the update, a couple more comments that I missed the first time. Squashing the changes is fine.

jbondpdx commented:
Thank you, @domcleal , I've updated the PR.

domcleal commented:
Merged, thanks @jbondpdx.
